src/main/webapp/components/UploadForm.vue

```diff
@@ -0,0 +1,131 @@
+<template>
+  <v-form ref="form" @submit.prevent="onSubmit">
+    <v-row>
+      <v-col>
+        <v-file-input label="Upload a PDF"
+                      v-model="file"
+                      accept=".pdf"
+                      show-size
+                      @change="onFileChange">
+        </v-file-input>
+
+        <v-text-field label="Title"
+                      v-model="title"
+                      clearable>
+        </v-text-field>
+
+        <v-combobox label="School"
+                    v-model="school"
+                    ref="school"
+                    clearable chips
+                    :items="commonSchools">
+          <template v-slot:no-data>
+            <span v-html="noDataHtml"></span>
+          </template>
+        </v-combobox>
+
+        <v-combobox label="Course"
+                    v-model="course"
+                    ref="course"
+                    clearable chips
+                    :items="commonCourses">
+          <template v-slot:no-data>
+            <span v-html="noDataHtml"></span>
+          </template>
+        </v-combobox>
+      </v-col>
+
+      <v-col>
+        <iframe :src="previewUrl"></iframe>
+      </v-col>
+    </v-row>
+
+    <v-combobox label="Miscellaneous Labels"
+                v-model="miscLabels"
+                multiple chips deletable-chips
+                clearable hide-selected
+                :items="commonLabels">
+      <template v-slot:no-data>
+        <span v-html="noDataHtml"></span>
+      </template>
+    </v-combobox>
+
+    <v-btn type="submit">Post Note</v-btn>
+  </v-form>
+</template>
+
+<script>
+module.exports = {
+  data: function() {
+    return {
+      uploadUrl: null,
+      file: null,
+      previewUrl: null,
+      title: null,
+      school: "Unaffiliated",
+      commonSchools: ['UF', 'MIT', 'NYU', 'UCLA'],
+      course: "Unaffiliated",
+      commonCourses: ['CIS4301', 'ENC1101', 'PHY2049'],
```

Comment:
Once this is connected to the database, will you need to add a function to retrieve the values for variables like `commonCourses` and `commonLabels`?

Reply:
Yep! I think there's even a way to show a loading animation while the query results come in, or load them early (eager).

Reply:
Ok got it, do you think you could add the TODOs into the code as well?